File: Dimatica/Dimatica/Services/DutyService.cs
using Dimatica.Entities.DTOs;
using Dimatica.Entities.DTOs.Duties;
using Dimatica.Entities.Interfaces.Repositories;
using Dimatica.Entities.Interfaces.Services;
using System.Collections.Generic;

namespace Dimatica.Services
{
    public class DutyService: IDutyService
    {
        private readonly IDutyRepository _dutyRepository;
        public DutyService(
            IDutyRepository dutyRepository
        ) 
        {
            _dutyRepository = dutyRepository;
        }

        public DutyDTO Create(DutyDTO dto)
        {
            _dutyRepository.Create(dto);
        }

        public bool Delete(DutyDTO dto)
        {
            throw new System.NotImplementedException();
        }

        public List<DutyDTO> Retrieve(PaginationDTO dto)
        {
            throw new System.NotImplementedException();
        }

        public DutyDTO Update(DutyDTO dto)
        {
            throw new System.NotImplementedException();
        }
    }
}

File: Dimatica/Dimatica/ClientApp/src/app/duty/services/duty.service.ts
import { Injectable } from '@angular/core';
import { DutyApiService } from '../../services/duty.service';
import { DutyDto } from '../interfaces/duty.dto.interface';

@Injectable({
  providedIn: 'root'
})
export class DutyService {

  constructor(private dutyService: DutyApiService) { }

  public async get(): Promise<DutyDto[]> {
    return await this.dutyService.get();
  }

  public async create(body: DutyDto): Promise<DutyDto> {
    return await this.dutyService.create(body);
  }

  public async update(body: DutyDto): Promise<DutyDto> {
    return await this.dutyService.update(body);
  }

  public async delete(id: string): Promise<boolean> {
    return await this.dutyService.delete(id);
  }
}
